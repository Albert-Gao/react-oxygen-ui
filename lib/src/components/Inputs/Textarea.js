"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const Input_1 = require("./Input");
exports.Textarea = props => (<Input_1.Input as="textarea" {...props}/>);
