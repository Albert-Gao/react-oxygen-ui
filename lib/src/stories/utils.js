"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const addon_actions_1 = require("@storybook/addon-actions");
const addon_knobs_1 = require("@storybook/addon-knobs");
const styles_1 = require("../styles");
function arrayToObject(data, keyToMap) {
    if (!data || data.length === 0) {
        return {};
    }
    const result = {};
    data.forEach((item, index) => {
        const i = keyToMap ? keyToMap[index] : index;
        result[i] = item;
    });
    return result;
}
exports.arrayToObject = arrayToObject;
// We wrap the action to prevent been display in addon-jsx
function wrapAction(actionName) {
    const wrapped = addon_actions_1.action(actionName);
    wrapped.toString = () => `action('${actionName}')`;
    return wrapped;
}
exports.wrapAction = wrapAction;
const keyOfColors = Object.keys(styles_1.defaultTheme.colors);
exports.Knobs = {
    spacesKnob: (propName, defaultValue = 2) => addon_knobs_1.select(propName, styles_1.DEFAULT_SPACE, defaultValue),
    colorKnob: (propName, defaultValue = 'primary') => addon_knobs_1.select(propName, arrayToObject(keyOfColors, keyOfColors), defaultValue),
    fontWeightKnob: (defaultValue = 'regular') => addon_knobs_1.select('fontWeight', {
        thin: 'thin',
        light: 'light',
        regular: 'regular',
        medium: 'medium',
        bold: 'bold',
        black: 'black',
    }, defaultValue),
    fontSizeKnob: (propName = 'fontSize', defaultValue = 1) => addon_knobs_1.select(propName, styles_1.DEFAULT_FONT_SIZE, defaultValue),
    sizeKnob: (propName, defaultValue = 'medium') => addon_knobs_1.select(propName, styles_1.DEFAULT_SIZE, styles_1.DEFAULT_SIZE[defaultValue]),
};
