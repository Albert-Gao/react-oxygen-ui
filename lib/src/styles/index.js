"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var fontSpecs_1 = require("./theme/constants/fontSpecs");
exports.DEFAULT_FONT_SIZE = fontSpecs_1.DEFAULT_FONT_SIZE;
exports.fontSpecs = fontSpecs_1.fontSpecs;
var spaces_1 = require("./theme/constants/spaces");
exports.DEFAULT_SPACE = spaces_1.DEFAULT_SPACE;
var sizes_1 = require("./theme/constants/sizes");
exports.DEFAULT_SIZE = sizes_1.DEFAULT_SIZE;
var colors_1 = require("./theme/constants/colors");
exports.DEFAULT_COLORS = colors_1.DEFAULT_COLORS;
var theme_1 = require("./theme/theme");
exports.defaultTheme = theme_1.defaultTheme;
__export(require("./theme/theme.utils"));
var styled_1 = require("./styled");
exports.styled = styled_1.styled;
exports.css = styled_1.css;
exports.createGlobalStyle = styled_1.createGlobalStyle;
exports.keyframes = styled_1.keyframes;
exports.ThemeProvider = styled_1.ThemeProvider;
var utils_1 = require("./utils");
exports.getTransitionValue = utils_1.getTransitionValue;
exports.getSafeValue = utils_1.getSafeValue;
exports.getSafeCss = utils_1.getSafeCss;
