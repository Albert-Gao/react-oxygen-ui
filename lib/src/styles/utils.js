"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const styled_system_1 = require("styled-system");
const utils_1 = require("../utils/utils");
exports.getMediaQuery = (screenSize, css) => `${styled_system_1.createMediaQuery(screenSize)} {
    ${css};
  }`;
exports.getTransitionValue = (propertyNames = 'transform') => `${propertyNames} 0.3s cubic-bezier(0.53, 0.01, 0.36, 1.63);`;
function getSafeValue(target, defaultValue) {
    return utils_1.is.notExist(target) ? defaultValue : target;
}
exports.getSafeValue = getSafeValue;
function getSafeCss(target) {
    return getSafeValue(target, '');
}
exports.getSafeCss = getSafeCss;
