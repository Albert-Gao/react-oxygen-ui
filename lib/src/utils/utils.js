"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.is = {
    function: (v) => typeof v === 'function',
    number: (v) => typeof v === 'number',
    notExist: (v) => v === undefined || v === null,
};
exports.getRefProp = (takeRef) => takeRef
    ? {
        ref: takeRef,
    }
    : {};
