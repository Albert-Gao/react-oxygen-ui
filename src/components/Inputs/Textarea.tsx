import React from 'react';
import { IInput, Input } from './Input';

export const Textarea: React.FC<IInput> = props => (
  <Input as="textarea" {...props} />
);
