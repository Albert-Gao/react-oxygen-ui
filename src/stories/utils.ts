import { action } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { Responsive } from '../components';
import {
  ColorType,
  DEFAULT_FONT_SIZE,
  DEFAULT_SIZE,
  DEFAULT_SPACE,
  defaultTheme,
  FontWeightType,
} from '../styles';

export function arrayToObject<T>(
  data: T[],
  keyToMap?: string[],
): { [id: string]: T } {
  if (!data || data.length === 0) {
    return {};
  }

  const result: { [id: string]: T } = {};
  data.forEach((item, index) => {
    const i = keyToMap ? keyToMap[index] : index;
    result[i] = item;
  });

  return result;
}

// We wrap the action to prevent been display in addon-jsx
export function wrapAction(actionName: string) {
  const wrapped = action(actionName);
  wrapped.toString = () => `action('${actionName}')`;

  return wrapped;
}

const keyOfColors = Object.keys(defaultTheme.colors) as ColorType[];

export const Knobs = {
  spacesKnob: (propName: string, defaultValue: number = 2) =>
    select<number>(propName, DEFAULT_SPACE, defaultValue),

  colorKnob: (
    propName: string,
    defaultValue: ColorType = 'primary',
  ) =>
    select<string>(
      propName,
      arrayToObject<string>(
        keyOfColors as string[],
        keyOfColors as string[],
      ),
      defaultValue,
    ),

  fontWeightKnob: (defaultValue: FontWeightType = 'regular') =>
    select(
      'fontWeight',
      {
        thin: 'thin',
        light: 'light',
        regular: 'regular',
        medium: 'medium',
        bold: 'bold',
        black: 'black',
      },
      defaultValue,
    ),

  fontSizeKnob: (
    propName: string = 'fontSize',
    defaultValue: Responsive = 1,
    // @ts-ignore
  ) => select(propName, DEFAULT_FONT_SIZE, defaultValue),

  sizeKnob: (
    propName: string,
    defaultValue: keyof typeof DEFAULT_SIZE = 'medium',
  ) => select(propName, DEFAULT_SIZE, DEFAULT_SIZE[defaultValue]),
};
