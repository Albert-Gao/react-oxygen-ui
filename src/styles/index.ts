import { Responsive } from '../components';
import { defaultTheme } from './theme/theme';


export type ThemeType = typeof defaultTheme;
export type SpaceType = Responsive;
export type FontSizeType = Responsive;
export type GradientType = keyof ThemeType['gradients'];
export type ColorType = string | string[];
export type FontWeightType = keyof ThemeType['fontWeights'];
export type BoxShadowType = number | number[];

export { DEFAULT_FONT_SIZE, fontSpecs } from './theme/constants/fontSpecs';
export { DEFAULT_SPACE } from './theme/constants/spaces';
export { DEFAULT_SIZE } from './theme/constants/sizes';
export { DEFAULT_COLORS } from './theme/constants/colors';
export { defaultTheme } from './theme/theme';
export * from './theme/theme.utils';

export {
  styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} from './styled';

export {
  getTransitionValue,
  getSafeValue,
  getSafeCss,
} from './utils';
