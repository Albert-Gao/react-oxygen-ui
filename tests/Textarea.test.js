import "jest-styled-components";
import "jest-dom/extend-expect";
import React from "react";
import { render, cleanup } from "react-testing-library";
import { Textarea } from "../src/components/Inputs/Textarea";
import { defaultTheme } from "../src/styles/IThemeType";
import { TestThemeProvider as ThemeProvider } from "./utils";

afterEach(cleanup);

describe("Test <Input />: ", () => {
  test("css prop should be added", () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Textarea css={"background-color: black;"} />
      </ThemeProvider>
    ).container;
    expect(firstChild).toMatchSnapshot();

    // the new css is there
    expect(firstChild).toHaveStyleRule("background-color", "black");

    expect(firstChild.tagName).toEqual("TEXTAREA");
  });
});
