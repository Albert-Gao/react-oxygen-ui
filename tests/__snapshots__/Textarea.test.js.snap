// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Input />:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  background-color: black;
}

<textarea
  class="c0"
/>
`;
